### PR TITLE
PENG-158 UBI should not allow undefined commands

### PR DESCRIPTION
**Pull request by rohandesai**
- now validating commands per capability of the device in the smartapp

@jasonbio @larsfinander

### CODE REVIEW

File: smartapps/smartthings/ubi.src/ubi.groovy [hunk 1]
```diff
@@ -442,31 +442,90 @@ def executePhrase() {
...
 	def command = request.JSON?.command
-	if (command)
-	{
-		command = command.toLowerCase()
-		devices."$command"()
+	if (!devices) {
+		httpError(404, "Devices not found")
+	}
+	if (command){
+		//command = command.toLowerCase()
```
Comment (larsfinander) [inline smartapps/smartthings/ubi.src/ubi.groovy]: Leftover comment?

File: smartapps/smartthings/ubi.src/ubi.groovy [hunk 2]
```diff
@@ -442,31 +442,90 @@ def executePhrase() {
...
+	if (!device) {
 			httpError(404, "Device not found")
-		}
-		else
-		{
-			device."$command"()
-		}
 	}
+
+	//def command = params.command
```
Comment (larsfinander) [inline smartapps/smartthings/ubi.src/ubi.groovy]: leftover comments

File: smartapps/smartthings/ubi.src/ubi.groovy [hunk 3]
```diff
@@ -442,31 +442,90 @@ def executePhrase() {
...
+	if (command) {
+		//command = command.toLowerCase()
```
Comment (larsfinander) [inline smartapps/smartthings/ubi.src/ubi.groovy]: leftover comments

**Comment by larsfinander** [comment]
👍

**Comment by rohandesai** [comment]
Tested this with @igalburt , and as discussed with him this is good to go. Merging this.

**Comment by bflorian** [comment]
I don't see any problem with this change but it does seem to be duplicating much of the logic that is already in the platform in the [DeviceWrapper](https://github.com/PhysicalGraph/data-management/blob/master/src/groovy/physicalgraph/app/DeviceWrapper.groovy#L44-L67) class that will throw an error if a SmartApp tried to call a command that the device itself does not support. 

Can't this logic be simplified to just constrain the /switches path to only execute on and off commands, the /locks path to only support lock/unlock, etc., and leave the checking of wether the device itself supports a given command to the platform?

**Pull request merged**